src/store/reducers/causes.js

```diff
@@ -1,15 +1,49 @@
 import { ACTION_TYPES } from '../actionTypes';
 
 const initialState = {
-  data: {},
+  data: null,
   loading: false,
   error: false
 };
+const dataSanitization = data => {
+  if (data) {
+    const {
+      id,
+      causes,
+      causes_heading: { heading_primary, heading_secondary }
+    } = data;
+    const sanitizedCauses = causes.map(
+      ({ title, description, raised, goal, image: { url: imgURL }, id }) => {
+        return {
+          title,
+          description,
+          raised,
+          goal,
+          imgURL,
+          id
+        };
+      }
+    );
+    return {
+      id,
+      causes: sanitizedCauses,
+      causes_heading: {
+        heading_primary,
+        heading_secondary
+      }
+    };
+  }
+  else {
+    return null
+  }
+};
 
 export const causesReducer = (state = initialState, { type, payload }) => {
   switch (type) {
     case ACTION_TYPES.CAUSES.SET_DATA:
-      const newState = { ...state, data: payload };
+      let sanitizeData = dataSanitization(payload);
```

Comment:
this logic shouldn't be in the reducer, it should be in the action (before dispatching it)